junit-platform-commons/src/main/java/org/junit/platform/commons/support/conversion/ConversionSupport.java

```diff
@@ -97,48 +48,50 @@ private ConversionSupport() {
 	 * type is a reference type
 	 *
 	 * @since 1.11
+	 * @see DefaultConverter
+	 * @deprecated Use {@link #convert(Object, TypeDescriptor, ClassLoader)} instead.
 	 */
-	@SuppressWarnings("unchecked")
+	@Deprecated
+	@API(status = DEPRECATED, since = "5.13")
 	public static <T> T convert(String source, Class<T> targetType, ClassLoader classLoader) {
-		if (source == null) {
-			if (targetType.isPrimitive()) {
-				throw new ConversionException(
-					"Cannot convert null to primitive value of type " + targetType.getTypeName());
-			}
-			return null;
-		}
+		return convert(source, TypeDescriptor.forType(targetType), getClassLoader(classLoader));
```

Comment:
I wonder if this should delegate to the other `convert` method or should invoke `DefaultConverter` directly, skipping the new service loader logic. For now, I went with the former.